rust-runtime/smithy-client/src/hyper_impls.rs

```diff
@@ -64,6 +67,53 @@ impl HyperAdapter<()> {
     }
 }
 
+/// Downcast errors coming out of hyper into an appropriate `ConnectorError`
+fn downcast_error(err: BoxError) -> ConnectorError {
+    // is a `TimedOutError` (from smithy_async::timeout) in the chain? if it is, this is a timeout
+    if find_source::<TimedOutError>(err.as_ref()).is_some() {
+        return ConnectorError::timeout(err);
+    }
+    // is the top of chain error actually already a client error? return that directly
+    let err = match err.downcast::<ConnectorError>() {
+        Ok(client_error) => return *client_error,
+        Err(box_error) => box_error,
+    };
+    // generally, the top of chain will probably be a hyper error. Go through a set of hyper specific
+    // error classifications
+    let err = match err.downcast::<hyper::Error>() {
+        Ok(hyper_error) => return to_client_error(*hyper_error),
+        Err(box_error) => box_error,
+    };
+
+    // otherwise, we have no idea!
+    ConnectorError::other(err, None)
+}
+
+/// Convert a [`hyper::Error`] into a [`ConnectorError`]
+fn to_client_error(err: hyper::Error) -> ConnectorError {
```

Comment:
Take a pass on renaming `client_error` to `connector_error`.

Reply:
good call